security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -95,10 +83,18 @@ func (c *citadelClient) CSRSign(ctx context.Context, reqID string, csrPEM []byte
 		Csr:              string(csrPEM),
 		ValidityDuration: certValidTTLInSec,
 	}
+	if token != "" {
+		// add Bearer prefix, which is required by Citadel.
+		token = bearerTokenPrefix + token
+		ctx = metadata.NewOutgoingContext(ctx, metadata.Pairs("Authorization", token, "ClusterID", c.clusterID))
+	} else {
+		err := c.reconnect()
```

Comment:
I think the question is still not well answered, why do we need to reconnect in this case? if you look into the reconnect(), the code path is the same even if the token is not empty, so I wonder what is changed when you do the reconnect?

Reply:
the client will use the new cert that are rotated. So will include that into a integration test for testing the whole workflow logic

Reply:
Does it use token in the very first request to get the certificate and then in the following request switch to use certificate? (or we never use token and the cert is added manually out-of-bound?)